addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3212,6 +3212,41 @@ export class OdooEditor extends EventTarget {
                 list.replaceChildren(...rangeContent.childNodes);
                 rangeContent = list;
             }
+            if (rangeContent.firstChild.nodeName === 'TR' || rangeContent.firstChild.nodeName === 'TD') {
+                // We enter this case only if selection is within single table.
+                const table = closestElement(range.commonAncestorContainer, 'table');
+                const tableClone = table.cloneNode(true);
+                const isTableFullySelected =
+                    table.parentElement && closestElement(table.parentElement, 'td.o_selected_td') ||
+                    [...table.querySelectorAll('td')]
+                        .filter(td => closestElement(td, 'table') === table)
+                        .every(td => td.classList.contains('o_selected_td'));
```

Comment:
Maybe this would warrant a comment ? What about (break at 80 chars):
```js
// A table is considered fully selected if it is nested inside a cell that is itself selected, or if all its cells are selected.
```